api/src/main/scala/org/ensime/api/common.scala

```diff
@@ -93,8 +93,8 @@ object RefactorType {
 }
 
 final case class SourceFileInfo(
-    file: File,
-    contents: Option[String] = None,
+    file: EnsimeFile,
+    @deprecating("prefer contentsIn, contents will be removed") contents: Option[String] = None,
```

Comment:
But if you remove `contents` how will clients work with in-memory files? Right now this is used, for instance, to ask for completions or for type-check requests.